tools/autopatches/autopatch.py

```diff
@@ -18,16 +17,27 @@
     get_catalog,
 )
 
+
+APPS_CACHE_DIR = REPO_APPS_ROOT / ".apps_cache"
+
+
+def app_cache_folder(app: str) -> Path:
+    return APPS_CACHE_DIR / app
+
+
 my_env = os.environ.copy()
 my_env["GIT_TERMINAL_PROMPT"] = "0"
-os.makedirs(".apps_cache", exist_ok=True)
 
-login = (
-    (REPO_APPS_ROOT / "tools/.github_login").open("r", encoding="utf-8").read().strip()
-)
-token = (
-    (REPO_APPS_ROOT / "tools/.github_token").open("r", encoding="utf-8").read().strip()
-)
+try:
+    login = (
+        (REPO_APPS_ROOT / "tools/.github_login").open("r", encoding="utf-8").read().strip()
```

Comment:
It's a Path() instance, isn't it?
So:
`(REPO_APPS_ROOT / "tools/.github_login").read_text().strip()`
see: https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text

Reply:
Oh god that's an amazing api !